`examples/gtk` needs better error handling, maybe a README as well?

Looks like magic that this works. 

But before it worked for me, I got this when I run the example: 

```
 > make
/bin/sh: 1: pkg-config: not found
../../bin/fz -jvm \
	-JLibraries="" \
	./gtk.fz

error 1: java.lang.ExceptionInInitializerError
	at fzC_fuzion_runtime_init.fzRoutine({base.fum}/fuzion_runtime_init.fz)
	at fzC_universe.fzRoutine(--builtin--:1)
	at fzC_universe.fz_run(--builtin--)
	at dev.flang.be.jvm.runtime.FuzionThread.lambda$new$1(FuzionThread.java:120)
	at dev.flang.util.Errors.runAndExit(Errors.java:912)
	at dev.flang.be.jvm.runtime.FuzionThread.lambda$new$2(FuzionThread.java:132)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: java.lang.IllegalArgumentException: Cannot open library: lib.so
	at java.base/java.lang.foreign.SymbolLookup.libraryLookup(SymbolLookup.java:277)
	at java.base/java.lang.foreign.SymbolLookup.libraryLookup(SymbolLookup.java:238)
	at dev.flang.be.jvm.runtime.Runtime.get_method_handle(Runtime.java:1396)
	at fzC_fzE_init.<clinit>({base.fum}/native.fz)
	... 7 more


*** fatal errors encountered, stopping.
one error.
make: *** [Makefile:31: jvm] Fehler 1
```

Would be nice if the Makefile would first check if `gtk` is installed and prodice a better error message and suggestion how to install. 

Maybe all the examples should come with a `readme.md` explaining what they do.

--- a/examples/gtk/Makefile
+++ b/examples/gtk/Makefile
@@ -24,7 +24,15 @@
 # To run this example, you must install gtk libs.
 # e.g.: apt-get install libgtk-4-dev
 
-jvm:
+
+.PHONY: jvm c compile test
+
+# Test if gtk4 and pkg-config are installed
+test:
+	@command -v pkg-config >/dev/null 2>&1 || { echo >&2 "Error: pkg-config is not installed."; exit 1; }
+	@pkg-config --exists gtk4 || { echo >&2 "Error: gtk4 is not found via pkg-config."; exit 1; }
+
+jvm: test
 	../../bin/fz -jvm \
 		-JLibraries="$(shell pkg-config --libs gtk4 | sed 's/-l//g')" \
 		./gtk.fz
@@ -32,7 +40,7 @@ jvm:
 c: compile
 	./gtk_ex
 
-compile:
+compile: test
 	../../bin/fz -c \
 		-CFlags="$(shell pkg-config --cflags gtk4) -Wno-incompatible-function-pointer-types" \
 		-CInclude=gtk/gtk.h \

--- a/examples/gtk/gtk.fz
+++ b/examples/gtk/gtk.fz
@@ -21,6 +21,7 @@
 #
 # -----------------------------------------------------------------------
 
+# Definitions of native functions
 
 gtk_application_window_new(gtkApplication Native_Ref) Native_Ref => native
 gtk_window_set_title(gtkWidget Native_Ref, c_str array u8) unit => native
@@ -32,6 +33,16 @@ g_application_run(app Native_Ref, argc i32, argv Native_Ref) i32 => native
 g_object_unref(app Native_Ref) unit => native
 
 
+# minimal example on how to use GTK from fuzion
+#
+# to run or compile use:
+#
+#     make jvm
+#
+# or
+#
+#     make c
+#
 gtk =>
 
   activate : Function unit Native_Ref Native_Ref is

